### ISSUE TITLE
Find solution for needed regcode parsing

### ISSUE THREAD

**Issue by saschagrunert** [SUSE/container-suseconnect#28]
The following code part decides whether a registration code retrieval is needed or not:

https://github.com/SUSE/container-suseconnect/blob/937fa64e4202452aa6041b74efece1d1bcd3835f/internal/products.go#L121-L129

This is currently a bit unflexible and proxies for the server `https://scc.suse.com` won't work any more since it will be assumed that no registration code is needed. A solution would be to make it configurable, but it has to be evaluated how and where.

**Comment by flavio** [comment]
I would expect users living in environments with http/https proxies would have to pass some environment variables to the container to make suse-connect work.

Are you taking about http proxies?

**Comment by saschagrunert** [comment]
> I would expect users living in environments with http/https proxies would have to pass some environment variables to the container to make suse-connect work.
> 
> Are you taking about http proxies?

Yes, we could also add some automatism, since the error messages from the server indicate what needs to be done next:

* response to request *without* credentials provided, but needed:
```json
{
  "type": "error",
  "error": "Please provide system/organization credentials.",
  "localized_error": "Please provide system/organization credentials."
}

```
* response to request *with wrong* credentials provided:
```json
{
  "type": "error",
  "error": "Invalid system credentials",
  "localized_error": "Invalid system credentials"
}
```

So we could add an additional path parsing the error and then decide whether an additional request with previously `parsedRegcodes` is needed.

**Comment by saschagrunert** [comment]
Closed with #30
